types/mapcmd/put-all.go

```diff
@@ -0,0 +1,217 @@
+
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/types"
+	"github.com/spf13/cobra"
+
+	hzcerror "github.com/hazelcast/hazelcast-commandline-client/errors"
+	"github.com/hazelcast/hazelcast-commandline-client/internal"
+	fds "github.com/hazelcast/hazelcast-commandline-client/types/flagdecorators"
+)
+
+func NewPutAll(config *hazelcast.Config) *cobra.Command {
+	var (
+		entries []types.Entry
+	)
+	var (
+		jsonEntryPath string
+		jsonEntries   map[string]interface{}
+	)
+	var (
+		mapName string
+		mapKeys,
+		mapValues,
+		mapValueTypes,
+		mapValueFiles []string
+	)
+	validateJsonEntryFlag := func() error {
+		if (len(mapKeys) |
+			len(mapValues) |
+			len(mapValueTypes) |
+			len(mapValueFiles)) != 0 {
+			return hzcerror.NewLoggableError(nil, fmt.Sprintf("%s is already set, there cannot be additional flags", fds.JsonEntryFlag))
+		}
+		return nil
+	}
+	validateValuesFlag := func() ([]byte, []int, error) {
+		valueCount := len(mapValues) + len(mapValueFiles)
+		if valueCount != len(mapKeys) {
+			return nil, nil, hzcerror.NewLoggableError(nil, "number of keys and values does not match")
+		}
+		vOrder, tOrder := ObtainOrderingOfValueFlags(os.Args)
+		if vOrder == nil {
+			return nil, nil, hzcerror.NewLoggableError(nil, "correct order of values cannot be taken")
+		}
+		return vOrder, tOrder, nil
+	}
+	executePutAll := func(ctx context.Context, cmd *cobra.Command, m *hazelcast.Map, entries []types.Entry) error {
+		var err error
+		err = m.PutAll(ctx, entries...)
+		if err != nil {
+			cmd.Println("Cannot put given entries")
+			isCloudCluster := config.Cluster.Cloud.Enabled
+			if networkErrMsg, handled := internal.TranslateNetworkError(err, isCloudCluster); handled {
+				err = hzcerror.NewLoggableError(err, networkErrMsg)
+			}
+			return err
+		}
+		return nil
+	}
+	cmd := &cobra.Command{
+		Use:   "put-all [--name mapname | {[[--key keyname]... | [[--value-file file | --value value][--value-type type]]...] | [--json-entry jsonEntryFile]}]",
+		Short: "Put values to map",
+		Example: `  # Put key, value pairs to map.
+  hzc map put-all -n mapname -k k1 -v v1 -k k2 -v v2
+  
+  # Put key, value pairs to map in another order.
+  hzc map put-all -n mapname -k k1 -k k2 -v v1 -v v2
+  
+  # Put key, value pairs to map but one of the value type is json file.
+  hzc map put-all -n mapname -k k1 -f valueFile.json -t json -k k2 -v v2
+
+  # Put all key, value pairs to map from the entry json file .
+  hzc map put-all -n mapname --json-entry entries.json
+
+  # Example json entry file
+  {
+    "key1": "value1",
+    "key2": {
+      "innerData": "data",
+      "anotherInnerData": 5.0
+    },
+    "key3": true,
+    "key4": [1, 2, 3, 4, 5]
+  }
+  - Entries with "null" values are being ignored.
+  
+  # Coupling rule of keys and values given in different order
+  - Keys and values are being coupled according to first given first matched manner. That means given first key will be matched with given first 
```

Comment:
```suggestion
  - Keys and values are coupled according to the order they are provided. That means the first given key will be matched with the first given 
```
I tried something but not sure if it is clearer. Wdyt?

Reply:
Changed! [2b8146a](https://github.com/hazelcast/hazelcast-commandline-client/pull/67/commits/2b8146aa690a38f5b06ee844b7122731324c4f5e)